cpp/src/strings/split/split_record.cu

```diff
@@ -49,7 +49,7 @@ std::unique_ptr<column> split_record_fn(strings_column_view const& input,
   if (input.is_empty()) { return make_empty_column(type_id::LIST); }
   if (input.size() == input.null_count()) {
     auto offsets = std::make_unique<column>(input.offsets(), stream, mr);
-    auto results = std::make_unique<column>(input.parent(), stream, mr);
+    auto results = make_empty_column(type_id::STRING);
```

Comment:
Sorry, after thinking about this more carefully I see a problem: In case the `input` column has non-empty nulls, this will cause data corruption as the `results` column is empty but `offsets` contains non-zero values copied from `input`. Thus, we should have either:
 * `results` is copied from `input.parent()` as your previous code, or
 * `offsets` is created from scratch with all `0` values.

Reply:
Right. The offsets would already by all zeroes if every entry in the input was all nulls. 
But the child column can be empty because, by definition, the child data should never be accessed for any null rows.
So technically, the child column could be anything since so no code should ever read it but any data in the child column of an all-null parent column could be considered as unsanitized.